dawn/src/dawn/SIR/SIR.h

```diff
@@ -330,28 +330,33 @@ struct Value : NonCopyable {
     return valueJson;
   }
 
-  template <typename T> struct TypeInfo {};
+  template <typename T>
+  struct TypeInfo {};
```

Comment:
I think there's no need for the definition here (it just obscures the error message slightly)

Reply:
It needs to be forward declared before it can be specialized outside the class. I made a larger change in the latest commit.

Reply:
Yes, but only declared, not defined
```
template <typename>
struct TypeInfo;
```
should do it I think.

Reply:
Ah, ok I see what you mean!